src/transactionsFactories/tokenManagementTransactionsFactory.ts

```diff
@@ -640,6 +660,231 @@ export class TokenManagementTransactionsFactory {
         }).build();
     }
 
+    createTransactionForModifyingRoyalties(options: {
+        sender: IAddress;
+        tokenIdentifier: string;
+        tokenNonce: bigint;
+        royalties: bigint;
+    }): Transaction {
+        const dataParts = [
+            "ESDTModifyRoyalties",
+            ...this.argSerializer.valuesToStrings([
+                new StringValue(options.tokenIdentifier),
+                new BigUIntValue(options.tokenNonce),
+                new BigUIntValue(options.royalties),
+            ]),
+        ];
+
+        return new TransactionBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: options.sender,
+            dataParts: dataParts,
+            gasLimit: this.config.gasLimitEsdtModifyRoyalties,
+            addDataMovementGas: true,
+        }).build();
+    }
+
+    createTransactionForSettingNewUris(options: {
+        sender: IAddress;
+        tokenIdentifier: string;
+        tokenNonce: bigint;
+        uris: string[];
+    }): Transaction {
+        if (!options.uris.length) {
+            throw new ErrBadUsage("");
+        }
+
+        const dataParts = [
+            "ESDTSetNewURIs",
+            ...this.argSerializer.valuesToStrings([
+                new StringValue(options.tokenIdentifier),
+                new BigUIntValue(options.tokenNonce),
+                ...options.uris.map((uri) => new StringValue(uri)),
+            ]),
+        ];
+
+        return new TransactionBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: options.sender,
+            dataParts: dataParts,
+            gasLimit: this.config.gasLimitSetNewUris,
+            addDataMovementGas: true,
+        }).build();
+    }
+
+    createTransactionForModifyingCreator(options: {
```

Comment:
Who is the new creator?

Reply:
the one who now owns the token and has the `ESDTRoleModifyCreator` role.